### PR TITLE
TST: Test to verify behavior of groupby.std with no numeric columns and numeric_only=True

### LINKED ISSUE #51080: BUG: groupby.std with no numeric columns and numeric_only=True raises
```
df = pd.DataFrame({'a': list('xyz'), 'b': list('def')})
gb = df.groupby('a')
print(gb.std(numeric_only=True))
```

raises `TypeError: All columns were dropped in grouped_reduce`. Instead, it should return an empty frame with index consisting of the groups.

### PR DESCRIPTION
**Pull request by numanijaz**
- [x] closes #51080
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -2843,3 +2843,22 @@ def test_obj_with_exclusions_duplicate_columns():
     result = gb._obj_with_exclusions
     expected = df.take([0, 2, 3], axis=1)
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_numeric_only_std_no_result():
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_groupby.py:2848]: Can you parameterize this test with `numeric_only` being True / False using `@pytest.mark.parametrize`.

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -2843,3 +2843,22 @@ def test_obj_with_exclusions_duplicate_columns():
...
+    # GH 51080
+    dicts_non_numeric = [{"a": "foo", "b": "bar"}, {"a": "car", "b": "dar"}]
+    df = DataFrame(dicts_non_numeric)
+    dfgb = df.groupby("a")
+    result = dfgb.std(numeric_only=True)
+
+    assert result.empty
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_groupby.py:2855]: Can you assert the full result here - construct `expected = DataFrame(...)` and then use `tm.assert_frame_equals(result, expected)`.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: pandas/core/groupby/groupby.py [hunk 3]
```diff
@@ -1915,7 +1915,7 @@ def std(
             .. versionadded:: 1.4.0
 
         numeric_only : bool, default False
-            Include only `float`, `int` or `boolean` data.
+            Include only `float`, `int` or `boolean` columns.
```
[the remark this answers is not in the document]
Comment (NumanIjaz) [reply pandas/core/groupby/groupby.py]: Makes sense. I have reverted these changes. But should't we have separate docstrings for DataFrameGroupBy and SeriesGroupBy for more clarity?

File: pandas/tests/groupby/test_groupby.py (hunk 1 above)
Comment (NumanIjaz) [reply pandas/tests/groupby/test_groupby.py]: Done.

File: pandas/tests/groupby/test_groupby.py (hunk 2 above)
Comment (NumanIjaz) [reply pandas/tests/groupby/test_groupby.py]: Done.

File: pandas/core/groupby/groupby.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rhshadrach) [reply pandas/core/groupby/groupby.py]: We could template these docstrings or just separate them out entirely; but that should be a separate PR.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @NumanIjaz!

**Pull request merged**